pkg/dbt_cloud/webhook.go

```diff
@@ -0,0 +1,140 @@
+package dbt_cloud
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type WebhookResponse struct {
+	Data   WebhookRead    `json:"data"`
+	Status ResponseStatus `json:"status"`
+}
+
+type WebhookRead struct {
+	WebhookId         string   `json:"id"`
+	Name              string   `json:"name"`
+	Description       string   `json:"description,omitempty"`
+	ClientUrl         string   `json:"client_url"`
+	EventTypes        []string `json:"event_types,omitempty"`
+	JobIds            []string `json:"job_ids,omitempty"`
+	Active            bool     `json:"active,omitempty"`
+	HmacSecret        *string  `json:"hmac_secret,omitempty"`
+	HttpStatusCode    *string  `json:"http_status_code,omitempty"`
+	AccountIdentifier *string  `json:"account_identifier,omitempty"`
+}
+
+type WebhookWrite struct {
+	WebhookId   string   `json:"id"`
+	Name        string   `json:"name"`
+	Description string   `json:"description,omitempty"`
+	ClientUrl   string   `json:"client_url"`
+	EventTypes  []string `json:"event_types,omitempty"`
+	JobIds      []int    `json:"job_ids,omitempty"`
+	Active      bool     `json:"active,omitempty"`
+}
+
+func (c *Client) GetWebhook(webhookID string) (*WebhookRead, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/v3/accounts/%s/webhooks/subscription/%s", c.HostURL, strconv.Itoa(c.AccountID), webhookID), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	webhookResponse := WebhookResponse{}
+	err = json.Unmarshal(body, &webhookResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return &webhookResponse.Data, nil
+}
+
+func (c *Client) CreateWebhook(
+	webhookId string,
+	name string,
+	description string,
+	clientUrl string,
+	eventTypes []string,
+	jobIds []int,
+	active bool,
+) (*WebhookRead, error) {
+
+	newWebhook := WebhookWrite{
+		WebhookId:   webhookId,
```

Comment:
shouldn't this not exist until after creation?

Reply:
It is actually required but with an empty value for the creation of the webhook.

It is set up empty [here](https://github.com/b-per/terraform-provider-dbt-cloud/blob/2e132ec6690c6b8b3f57a4add47de3e98dff52be/pkg/resources/webhook.go#L155)